RealmBrowser/Support/RLMModelExporter.m

```diff
@@ -569,7 +581,104 @@ + (NSString *)javaScriptDefinitionForProperty:(RLMProperty *)property
     }
     
     return definition;
-    
+}
+
+#pragma mark - Private methods - C# helpers
+
++ (NSArray *)cSharpModelsOfSchemas:(NSArray *)schemas withFileName:(NSString *)fileName {
+    NSMutableString *contents = [NSMutableString stringWithString:@"using System;\nusing System.Collections.Generic;\nusing Realms;\n\n"];
+
+    for (RLMObjectSchema *schema in schemas) {
+        [contents appendFormat:@"public class %@ : RealmObject\n{\n", schema.className];
+
+        for (RLMProperty *property in schema.properties) {
+            [contents appendFormat:@"%@\n\n", [[self cSharpDefinitionForProperty:property] indentedBy:4]];
+        }
+
+        // Delete extra newline
+        [contents deleteCharactersInRange:NSMakeRange(contents.length - 1, 1)];
+
+        [contents appendString:@"}\n\n"];
+    }
+
+    // Delete extra newline
+    [contents deleteCharactersInRange:NSMakeRange(contents.length - 1, 1)];
+
+    return @[@[[fileName stringByAppendingPathExtension:@"cs"], contents]];
+}
+
++ (NSString *)cSharpDefinitionForProperty:(RLMProperty *)property {
+    NSMutableString *definition = [NSMutableString string];
+
+    BOOL(^typeOptionalByDefault)(RLMPropertyType) = ^BOOL(RLMPropertyType type) {
+        switch (type) {
+            case RLMPropertyTypeString:
+            case RLMPropertyTypeData:
+            case RLMPropertyTypeObject:
+                return true;
+            default:
+                return false;
+        }
+    };
+
+    NSString *(^valueType)(NSString *, BOOL) = ^NSString *(NSString *typename, BOOL optional) {
+        return [typename stringByAppendingString:optional ? @"?" : @""];
+    };
+
+    if (property.isPrimary) {
+        [definition appendString:@"[PrimaryKey]\n"];
+    }
+
+    if (property.type == RLMPropertyTypeLinkingObjects) {
+        [definition appendFormat:@"[Backlink(nameof(%@.%@))]\n", property.objectClassName, property.linkOriginPropertyName];
+    }
+
+    if (!property.optional && typeOptionalByDefault(property.type)) {
+        [definition appendString:@"[Required]\n"];
```

Comment:
Right. It's a little confusing that it is always optional, yet it is in the `typeOptionalByDefault` collection that is only used to add an attribute for required properties (which it can never be). But I guess it's fine to leave it as is for future-proofness in case someone uses the `typeOptionalByDefault` collection.